From: Alex Merry <alex.merry@kde.org>
Date: Tue, 15 Dec 2015 19:26:47 +0000
Subject: Revert "backport commit b72fc5e56579035bf987075e16324ef95ef8e3d4"
X-Git-Url: http://quickgit.kde.org/?p=kdelibs.git&a=commitdiff&h=a02df05e4bd083f98147c86f88da2f818fc6c9f4
---
Revert "backport commit b72fc5e56579035bf987075e16324ef95ef8e3d4"

This reverts commit 4f7ea2f770cf062ef22293fbb21a086f3e0cbfcb.

This change seems to be causing more problems than it fixes - it's
probably just too big of a behaviour change for kdelibs. Which means
that akregator will probably keep randomly crashing, but the alternative
seems to be various other applications consistently crashing at exit.

If we can fix those applications (Kopete in particular), we can consider
re-applying this afterwards.

BUG: 355275
---


--- a/kparts/part.cpp
+++ b/kparts/part.cpp
@@ -350,7 +350,7 @@
     d->m_widget = 0;
     if (d->m_autoDeletePart) {
         kDebug(1000) << "deleting part" << objectName();
-        this->deleteLater();
+        delete this; // ouch, this should probably be deleteLater()
     }
 }
 

--- a/kparts/tests/parttest.cpp
+++ b/kparts/tests/parttest.cpp
@@ -48,7 +48,6 @@
     KParts::Part* part = new TestPart(0, 0);
     QPointer<KParts::Part> partPointer(part);
     delete part->widget();
-    QCoreApplication::sendPostedEvents(0, QEvent::DeferredDelete);
     QVERIFY(partPointer.isNull());
 }
 
@@ -58,7 +57,6 @@
     QPointer<KParts::Part> partPointer(part);
     QPointer<QWidget> widgetPointer(part->widget());
     delete part;
-    QCoreApplication::sendPostedEvents(0, QEvent::DeferredDelete);
     QVERIFY(widgetPointer.isNull());
 }
 

